### PR TITLE
chore: move type:check to lint hook

### LINKED ISSUE #3248: Alter`type:check` command
# Summary

- We currently do `type:check` in various hooks, which can slow down development, so we should move check to our `lint` hook.
- Potentially the best strategy for this issue is to roll back the [original PR](https://github.com/FuelLabs/fuels-ts/pull/3046), then re-add the checks to the related lint command.

## All hooks
- `pretest`
  - https://github.com/FuelLabs/fuels-ts/blob/next/apps/docs-snippets/package.json#L7
  - https://github.com/FuelLabs/fuels-ts/blob/next/internal/benchmarks/package.json#L9
  - https://github.com/FuelLabs/fuels-ts/blob/next/packages/fuel-gauge/package.json#L8
  
- `build`
  - https://github.com/FuelLabs/fuels-ts/blob/next/packages/create-fuels/package.json#L18

- `postbuild`.
  - https://github.com/FuelLabs/fuels-ts/blob/next/packages/fuels/package.json#L59

### PR DESCRIPTION
**Pull request by Rohit-Bhetal**
- Closes #3248

# Summary
This PR moves the TypeScript type checking from the various build-related scripts (pretest, post-build, etc.) to a dedicated `lint` script. This helps improve the development speed by reducing the number of redundant type checks performed during the build process.

# Checklist

- [x] All **changes** are **covered** by **tests** (or not applicable)
- [x] All **changes** are **documented** (or not applicable)
- [x] I **reviewed** the **entire PR** myself (preferably, on GH UI)
- [x] I **described** all **Breaking Changes** (or there's none)

### CODE REVIEW

**Review by Rohit-Bhetal — COMMENTED** [review]
Updated other package.json file also

**Review by maschad — DISMISSED** [review]
Thanks for your contribution @Rohit-Bhetal 👍🏾

**Review by Dhaiwat10 — DISMISSED** [review]

**Review by petertonysmith94 — DISMISSED** [review]
I believe we need to run `turbo run lint` from the root level package.json.

Maybe we could keep the `type:check` command hook for the project level `package.json`, and add the following on the root level:

```json
...,
"lint": "run-s type:check-tests lint:check prettier:check type:check",
"type:check": "turbo run type:check",
...,
```

**Comment by maschad** [comment]
> I believe we need to run `turbo run lint` from the root level package.json.
> 
> Maybe we could keep the `type:check` command hook for the project level `package.json`, and add the following on the root level:
> 
> ```json
> ...,
> "lint": "run-s type:check-tests lint:check prettier:check type:check",
> "type:check": "turbo run type:check",
> ...,
> ```

Right good catch @petertonysmith94 , if we are going that route then we will also have to add `type:check` to the [turbo tasks](https://github.com/FuelLabs/fuels-ts/blob/master/turbo.json#L32)

**Review by danielbate — DISMISSED** [review]
This PR needs a changeset, please run `pnpm changeset` from the root directory and commit the changeset.

**Review by Rohit-Bhetal — COMMENTED** [review]
Added a  type:check to turbo tasks and update root package.json lint script and made task to run the type check in turbo.json

**Comment by Rohit-Bhetal** [comment]
> This PR needs a changeset, please run `pnpm changeset` from the root directory and commit the changeset.

yep done

File: packages/create-fuels/package.json [hunk 1]
```diff
@@ -15,9 +15,9 @@
   },
   "license": "Apache-2.0",
   "scripts": {
-    "build": "run-s build:source type:check",
+    "build": "run-s build:source",
     "build:source": "tsup",
-    "type:check": "tsc --noEmit",
+    "lint": "tsc --noEmit",
```
Comment (petertonysmith94) [inline packages/create-fuels/package.json:20]:
```suggestion
    "type:check": "tsc --noEmit",
```

File: packages/fuel-gauge/package.json [hunk 2]
```diff
@@ -5,10 +5,10 @@
   "description": "",
   "author": "Fuel Labs <contact@fuel.sh> (https://fuel.network/)",
   "scripts": {
-    "pretest": "run-s build:forc build:process-predicates type:check",
+    "pretest": "run-s build:forc build:process-predicates",
     "build:forc": "pnpm fuels build",
     "build:process-predicates": "tsx ./scripts/process-predicates.ts",
-    "type:check": "tsc --noEmit"
+    "lint": "tsc --noEmit"
```
Comment (petertonysmith94) [inline packages/fuel-gauge/package.json:11]:
```suggestion
    "type:check": "tsc --noEmit"
```

File: packages/fuels/package.json [hunk 3]
```diff
@@ -56,9 +56,9 @@
     "build:package": "tsup",
     "build:browser": "pnpm vite build",
     "build:minified": "pnpm uglifyjs --compress --mangle --output dist/browser.min.mjs -- dist/browser.mjs",
-    "postbuild": "run-s type:declarations type:check",
+    "postbuild": "run-s type:declarations",
     "type:declarations": "tsc --emitDeclarationOnly -p tsconfig.dts.json",
-    "type:check": "tsc --noEmit",
+    "lint": "tsc --noEmit",
```
Comment (petertonysmith94) [inline packages/fuels/package.json:61]:
```suggestion
    "type:check": "tsc --noEmit",
```

File: internal/benchmarks/package.json [hunk 4]
```diff
@@ -5,8 +5,8 @@
     "dist"
   ],
   "scripts": {
-    "type:check": "tsc --noEmit",
-    "pretest": "run-s build:forc type:check",
+    "lint": "tsc --noEmit",
```
Comment (petertonysmith94) [inline internal/benchmarks/package.json:8]:
```suggestion
    "type:check": "tsc --noEmit",
```

File: apps/docs-snippets/package.json [hunk 5]
```diff
@@ -4,7 +4,8 @@
   "description": "",
   "private": true,
   "scripts": {
-    "pretest": "./scripts/pretest.sh"
+    "pretest": "./scripts/pretest.sh",
+    "lint": "tsc --noEmit"
```
Comment (petertonysmith94) [inline apps/docs-snippets/package.json:8]:
```suggestion
    "type:check": "tsc --noEmit"
```

File: .changeset/polite-seas-taste.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
+---
+"fuels": major
+---
+
+Move type-checking to a dedicated lint script across the project to improve development speed and reduce redundant checks
```
Comment (petertonysmith94) [inline .changeset/polite-seas-taste.md:5]:
```suggestion
chore: move type:check to lint hook
```

File: .changeset/polite-seas-taste.md [hunk 7]
```diff
@@ -0,0 +1,5 @@
+---
+"fuels": major
```
Comment (petertonysmith94) [inline .changeset/polite-seas-taste.md:2]:
```suggestion
```

This only affects development - therefore no release required.

File: package.json [hunk 8]
```diff
@@ -39,12 +39,12 @@
...
     "clinic:heap": "clinic heapprofiler --autocannon [ / ] -- node packages/${npm_config_package_name}/dist/profile.js",
     "test:integration": "vitest --run --config vitest.node.config.mts $(scripts/tests-find.sh --integration)",
-    "lint": "run-s type:check-tests lint:check prettier:check",
+    "lint": "run-s type:check-tests lint:check prettier:check type:check",
     "lint:check": "eslint . --ext .ts --max-warnings 0",
     "lint:fix": "pnpm lint:check --fix",
     "lint:md-links": "tsx ./scripts/lint-md-links",
     "lint:package-jsons": "tsx ./scripts/lint-package-jsons",
-    "type:check-tests": "tsc -p tsconfig.test.json",
+    "type:check": "turbo run type:check",
```
Comment (petertonysmith94) [inline package.json:47]:
```suggestion
    "type:check": "turbo run type:check",
    "type:check-tests": "tsc -p tsconfig.test.json",
```

**Review by petertonysmith94 — APPROVED** [review]

**Pull request merged**